app/src/main/java/de/grobox/transportr/map/PositionController.kt

```diff
@@ -78,10 +78,16 @@ import java.util.concurrent.TimeUnit
             PositionState.DISABLED
     }
 
+     private fun isProviderEnabled(provider: String): Boolean {
+         return locationManager.isProviderEnabled(provider)
+     }
+
     @RequiresPermission(ACCESS_FINE_LOCATION)
     override fun onActive() {
         for (provider in LOCATION_PROVIDERS) {
```

Comment:
```suggestion
        for (provider in LOCATION_PROVIDERS.filter(locationManager::isProviderEnabled)) {
```
and remove the extra if-statement and indentation. Slightly more idiomatic I'd say.

Reply:
And the extra private fun, obviously.